node.go

```diff
@@ -284,26 +284,30 @@ func (node *Node) writeBytes(w io.Writer) (n int, err error) {
 func (node *Node) set(t *Tree, key []byte, value []byte) (
 	newSelf *Node, updated bool, orphaned []*Node,
 ) {
+	version := t.version + 1
+
 	if node.isLeaf() {
 		switch bytes.Compare(key, node.key) {
 		case -1:
 			return &Node{
 				key:       node.key,
 				height:    1,
 				size:      2,
-				leftNode:  NewNode(key, value),
```

Comment:
Nice, so set version once on insert rather than that traversing callback craziness in Save.
That was super disconcerting. Glad you changed that,

:+1: